watcher/local/watcher_test.go

```diff
@@ -80,6 +85,204 @@ func Test_StartWatching(t *testing.T) {
 	})
 }
 
+func Test_Watcher_Working(t *testing.T) {
```

Comment:
I agree with @ggwpez , it takes some effort to understand the tests. In particular, it is not obvious what each test focuses on and what are the differences between them.

The test descriptions are helpful, but also not self explanatory, so some comments could probably help.

Reply:
In 67e4a04, added comments on top of each tests to explain what it focus on. 

In case, if you feel still any part of the tests is not self-explanatory / easy to comprehend, feel free to point out.

Reply:
@matthiasgeihs  In our discussion over call, we figured out that, it is particularly not obvious "how the expectations are set on the mocks". 

In fabd8a0, I have refactored the tests to address this, by
1. Renaming the `setRegistered` function to `setExpectionsRegisterCalls`, which sets the expectation for `Register` calls and
2. Adding another function `setExpectationSubscribeCall`, which sets the expectation for `Subscribe` calls.